packages/api/amplify_api_android/android/src/test/kotlin/com/amazonaws/amplify/amplify_api/GraphQLApiUnitTests.kt

```diff
@@ -48,15 +48,19 @@ const val underlyingMalformedException =
 
 const val underlyingInvalidApiException =
     "AmplifyException{message=The apiName request argument " +
-            "was not passed as a String, cause=java.lang.ClassCastException: class java.lang.Integer cannot be cast " +
-            "to class java.lang.String (java.lang.Integer and java.lang.String are in module java.base of loader 'bootstrap'), " +
+            "was not passed as a String, cause=java.lang.ClassCastException: class java.lang.Integer cannot be cast to " +
+            "class java.lang.String (java.lang.Integer and java.lang.String are in module java.base of loader 'bootstrap'), recoverySuggestion=The request should include the apiName as a String}"
+/*
```

Comment:
I know we've had some churn with these tests due to small difference in how the underlying libraries are returning the exceptions. Is that why this is commented out? Maybe as part of this PR we could just test for the relevant substring (or at least, I don't think we should leave a a commented out variant)?

Reply:
updated to use custom argument matcher.

  ```
      verify(mockResult).error(
                eq("ApiException"),
                eq(ExceptionMessages.defaultFallbackExceptionMessage),
                argThat(
                    InvalidApiMapErrorArgumentMatcher()
                )
        )
    }

    class InvalidApiMapErrorArgumentMatcher : ArgumentMatcher<Map<String, String>>{
        override fun matches(argument: Map<String, String>): Boolean {
            return argument["message"] == ExceptionMessages.missingExceptionMessage &&
            argument["recoverySuggestion"] == ExceptionMessages.missingRecoverySuggestion &&
            argument["underlyingException"]!!.contains(underlyingInvalidApiException)
        }
    }
```